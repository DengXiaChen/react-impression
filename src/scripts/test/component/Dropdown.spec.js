import React from 'react';
import { shallow, mount } from 'enzyme';
import { Dropdown, Button } from '../../components/impression';

const setup = () => {
    const props = {
        trigger: 'click',
        active: true,
    };
    const mockFn = jest.fn();

    const temp = (
        <Dropdown {...props}>
            <Dropdown.Trigger>
                <Button theme="primary">Dropdown</Button>
            </Dropdown.Trigger>
            <Dropdown.Menu>
                <Dropdown.MenuItem disabled onClick={mockFn}>
                    菜单一
                </Dropdown.MenuItem>
                <Dropdown.MenuItem onClick={mockFn}>
                    菜单二
                </Dropdown.MenuItem>
                <Dropdown.MenuItem onClick={mockFn}>
                    菜单三
                </Dropdown.MenuItem>
                <Dropdown.MenuDivider />
                <Dropdown.MenuItem onClick={mockFn}>
                    菜单四
                </Dropdown.MenuItem>
                <Dropdown.MenuItem onClick={mockFn}>
                    菜单五
                </Dropdown.MenuItem>
                <Dropdown.MenuItem onClick={mockFn}>
                    菜单六
                </Dropdown.MenuItem>
            </Dropdown.Menu>
        </Dropdown>
    );

    const wrapper1 = shallow(temp);
    const wrapper = mount(temp);

    return {
        props,
        mockFn,
        wrapper,
        wrapper1,
    };
};

describe('Dropdown', () => {
    const { wrapper, wrapper1, mockFn } = setup();

    it('render', () => {
        expect(wrapper.find('.dropdown.active').exists()).toBe(true);
        expect(wrapper.find('.dropdown-menu').exists()).toBe(true);
        expect(wrapper.find('.dropdown-item')).toHaveLength(6);
    });

    it('click', () => {
        wrapper.find('.dropdown-item').findWhere(n => n.text() === '菜单三').simulate('click');
        expect(mockFn).toHaveBeenCalledTimes(1);
    });

    it('toggle', () => {
        wrapper1.find('.dropdown.active').simulate('click');
        expect(wrapper1.find('button').exists()).toBe(false);
    });

    it('ui', () => {
        expect(wrapper1).toMatchSnapshot();
    });
});
