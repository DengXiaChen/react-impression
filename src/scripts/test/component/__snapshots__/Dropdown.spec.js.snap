// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown ui 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="dropdown active"
>
    <DropdownTrigger
        toggleMenu={[Function]}
        trigger="click"
    >
        <Button
            theme="primary"
        >
            Dropdown
        </Button>
    </DropdownTrigger>
    <DropdownMenu
        right={false}
        toggleMenu={[Function]}
        trigger="click"
    >
        <DropdownMenuItem
            disabled={true}
            onClick={[Function]}
        >
            菜单一
        </DropdownMenuItem>
        <DropdownMenuItem
            disabled={false}
            onClick={[Function]}
        >
            菜单二
        </DropdownMenuItem>
        <DropdownMenuItem
            disabled={false}
            onClick={[Function]}
        >
            菜单三
        </DropdownMenuItem>
        <DropdownMenuDivider />
        <DropdownMenuItem
            disabled={false}
            onClick={[Function]}
        >
            菜单四
        </DropdownMenuItem>
        <DropdownMenuItem
            disabled={false}
            onClick={[Function]}
        >
            菜单五
        </DropdownMenuItem>
        <DropdownMenuItem
            disabled={false}
            onClick={[Function]}
        >
            菜单六
        </DropdownMenuItem>
    </DropdownMenu>
</div>,
  "nodes": Array [
    <div
      className="dropdown active"
>
      <DropdownTrigger
            toggleMenu={[Function]}
            trigger="click"
      >
            <Button
                  theme="primary"
            >
                  Dropdown
            </Button>
      </DropdownTrigger>
      <DropdownMenu
            right={false}
            toggleMenu={[Function]}
            trigger="click"
      >
            <DropdownMenuItem
                  disabled={true}
                  onClick={[Function]}
            >
                  菜单一
            </DropdownMenuItem>
            <DropdownMenuItem
                  disabled={false}
                  onClick={[Function]}
            >
                  菜单二
            </DropdownMenuItem>
            <DropdownMenuItem
                  disabled={false}
                  onClick={[Function]}
            >
                  菜单三
            </DropdownMenuItem>
            <DropdownMenuDivider />
            <DropdownMenuItem
                  disabled={false}
                  onClick={[Function]}
            >
                  菜单四
            </DropdownMenuItem>
            <DropdownMenuItem
                  disabled={false}
                  onClick={[Function]}
            >
                  菜单五
            </DropdownMenuItem>
            <DropdownMenuItem
                  disabled={false}
                  onClick={[Function]}
            >
                  菜单六
            </DropdownMenuItem>
      </DropdownMenu>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <Dropdown
        active={true}
        trigger="click"
>
        <DropdownTrigger>
                <Button
                        theme="primary"
                >
                        Dropdown
                </Button>
        </DropdownTrigger>
        <DropdownMenu
                right={false}
        >
                <DropdownMenuItem
                        disabled={true}
                        onClick={[Function]}
                >
                        菜单一
                </DropdownMenuItem>
                <DropdownMenuItem
                        disabled={false}
                        onClick={[Function]}
                >
                        菜单二
                </DropdownMenuItem>
                <DropdownMenuItem
                        disabled={false}
                        onClick={[Function]}
                >
                        菜单三
                </DropdownMenuItem>
                <DropdownMenuDivider />
                <DropdownMenuItem
                        disabled={false}
                        onClick={[Function]}
                >
                        菜单四
                </DropdownMenuItem>
                <DropdownMenuItem
                        disabled={false}
                        onClick={[Function]}
                >
                        菜单五
                </DropdownMenuItem>
                <DropdownMenuItem
                        disabled={false}
                        onClick={[Function]}
                >
                        菜单六
                </DropdownMenuItem>
        </DropdownMenu>
</Dropdown>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Dropdown {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "hideOptionsHandle": [Function],
        "props": Object {
          "active": true,
          "children": Array [
            <DropdownTrigger>
              <Button
                            theme="primary"
              >
                            Dropdown
              </Button>
</DropdownTrigger>,
            <DropdownMenu
              right={false}
>
              <DropdownMenuItem
                            disabled={true}
                            onClick={[Function]}
              >
                            菜单一
              </DropdownMenuItem>
              <DropdownMenuItem
                            disabled={false}
                            onClick={[Function]}
              >
                            菜单二
              </DropdownMenuItem>
              <DropdownMenuItem
                            disabled={false}
                            onClick={[Function]}
              >
                            菜单三
              </DropdownMenuItem>
              <DropdownMenuDivider />
              <DropdownMenuItem
                            disabled={false}
                            onClick={[Function]}
              >
                            菜单四
              </DropdownMenuItem>
              <DropdownMenuItem
                            disabled={false}
                            onClick={[Function]}
              >
                            菜单五
              </DropdownMenuItem>
              <DropdownMenuItem
                            disabled={false}
                            onClick={[Function]}
              >
                            菜单六
              </DropdownMenuItem>
</DropdownMenu>,
          ],
          "trigger": "click",
        },
        "refs": Object {},
        "state": Object {
          "active": true,
        },
        "toggleOptionsHandle": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="dropdown active"
>
          <DropdownTrigger
                    toggleMenu={[Function]}
                    trigger="click"
          >
                    <Button
                              theme="primary"
                    >
                              Dropdown
                    </Button>
          </DropdownTrigger>
          <DropdownMenu
                    right={false}
                    toggleMenu={[Function]}
                    trigger="click"
          >
                    <DropdownMenuItem
                              disabled={true}
                              onClick={[Function]}
                    >
                              菜单一
                    </DropdownMenuItem>
                    <DropdownMenuItem
                              disabled={false}
                              onClick={[Function]}
                    >
                              菜单二
                    </DropdownMenuItem>
                    <DropdownMenuItem
                              disabled={false}
                              onClick={[Function]}
                    >
                              菜单三
                    </DropdownMenuItem>
                    <DropdownMenuDivider />
                    <DropdownMenuItem
                              disabled={false}
                              onClick={[Function]}
                    >
                              菜单四
                    </DropdownMenuItem>
                    <DropdownMenuItem
                              disabled={false}
                              onClick={[Function]}
                    >
                              菜单五
                    </DropdownMenuItem>
                    <DropdownMenuItem
                              disabled={false}
                              onClick={[Function]}
                    >
                              菜单六
                    </DropdownMenuItem>
          </DropdownMenu>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="dropdown active"
>
          <DropdownTrigger
                    toggleMenu={[Function]}
                    trigger="click"
          >
                    <Button
                              theme="primary"
                    >
                              Dropdown
                    </Button>
          </DropdownTrigger>
          <DropdownMenu
                    right={false}
                    toggleMenu={[Function]}
                    trigger="click"
          >
                    <DropdownMenuItem
                              disabled={true}
                              onClick={[Function]}
                    >
                              菜单一
                    </DropdownMenuItem>
                    <DropdownMenuItem
                              disabled={false}
                              onClick={[Function]}
                    >
                              菜单二
                    </DropdownMenuItem>
                    <DropdownMenuItem
                              disabled={false}
                              onClick={[Function]}
                    >
                              菜单三
                    </DropdownMenuItem>
                    <DropdownMenuDivider />
                    <DropdownMenuItem
                              disabled={false}
                              onClick={[Function]}
                    >
                              菜单四
                    </DropdownMenuItem>
                    <DropdownMenuItem
                              disabled={false}
                              onClick={[Function]}
                    >
                              菜单五
                    </DropdownMenuItem>
                    <DropdownMenuItem
                              disabled={false}
                              onClick={[Function]}
                    >
                              菜单六
                    </DropdownMenuItem>
          </DropdownMenu>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Dropdown
    active={true}
    trigger="click"
>
    <DropdownTrigger>
        <Button
            theme="primary"
        >
            Dropdown
        </Button>
    </DropdownTrigger>
    <DropdownMenu
        right={false}
    >
        <DropdownMenuItem
            disabled={true}
            onClick={[Function]}
        >
            菜单一
        </DropdownMenuItem>
        <DropdownMenuItem
            disabled={false}
            onClick={[Function]}
        >
            菜单二
        </DropdownMenuItem>
        <DropdownMenuItem
            disabled={false}
            onClick={[Function]}
        >
            菜单三
        </DropdownMenuItem>
        <DropdownMenuDivider />
        <DropdownMenuItem
            disabled={false}
            onClick={[Function]}
        >
            菜单四
        </DropdownMenuItem>
        <DropdownMenuItem
            disabled={false}
            onClick={[Function]}
        >
            菜单五
        </DropdownMenuItem>
        <DropdownMenuItem
            disabled={false}
            onClick={[Function]}
        >
            菜单六
        </DropdownMenuItem>
    </DropdownMenu>
</Dropdown>,
}
`;
